addons/survey/models/survey_survey.py

```diff
@@ -434,6 +495,70 @@ def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
             questions = questions & answer.predefined_question_ids
         return questions, page_or_question_id
 
+    # ------------------------------------------------------------
+    # SESSIONS MANAGEMENT
+    # ------------------------------------------------------------
+
+    def session_next_question(self):
+        """ Triggers the next question of the session.
+
+        We artificially add 2 seconds to the 'current_question_start_time' to account for server delay.
+        As the timing can influence the attendees score, we try to be fair with everyone by giving them
+        an extra few seconds before we start counting down.
+
+        Frontend should take the delay into account by displaying the appropriate animations. """
+
+        self.ensure_one()
+
+        if not self.question_ids:
+            return
+
+        if not self.session_current_question_id:
+            question = self.question_ids[0]
+        else:
+            questions = list(enumerate(self.question_ids))
+            current_question_index = questions.index(
+                next(question for question in questions if question[1] == self.session_current_question_id)
+            )
+            if current_question_index < len(self.question_ids) - 1:
+                question = self.question_ids[current_question_index + 1]
+            else:
+                question = self.session_current_question_id
+
+        # using datetime.datetime because we want the millis portion
+        now = datetime.datetime.now()
+        self.write({
+            'session_current_question_id': question.id,
+            'session_current_question_start_time': fields.Datetime.now() + relativedelta(seconds=2)
+        })
+        self.env['bus.bus'].sendone(self.access_token, {
+            'question_start': now.timestamp(),
+            'type': 'next_question'
+        })
+
+    def _prepare_ranking_values(self):
+        """  Prepare the necessary values to display a ranking of the attendees for this session.
+        The ranking is descending and takes the total of the attendee points up to the current question. """
+        self.ensure_one()
+
+        survey_results = self.env['survey.user_input.line'].read_group(
+            [('user_input_id', 'in', self.user_input_ids.ids), ('answer_score', '!=', 0)],
+            ['user_input_id', 'answer_score:sum'],
+            ['user_input_id'],
+            orderby="answer_score desc",
+            limit=25
+        )
+
+        nicknames_by_user_input = {
+            user_input.id: user_input.survey_user_nickname
+            for user_input in self.user_input_ids
```

Comment:
Mmh.
When the session was its own model, this was handled automatically.
Here it will mess up with my limit=25.
I could add a related stored field `test_entry` on `user_input_line` but it will add a lot of noise just for that.

(And according to spec, surveys will be duplicated before starting the session *sigh* ).

Reply:
What could go wrong ?

Reply:
Ok you won, I did it :D

Reply:
No wait I did not, this reply was for another comment (I need more coffee)